app/scripts/lib/multichain-api/provider-authorize/handler.js

```diff
@@ -73,26 +81,33 @@ export async function providerAuthorizeHandler(req, res, _next, end, hooks) {
       },
     );
     const permittedAccounts = getAccountsFromPermission(subjectPermission);
-    const { flattenedRequiredScopes, flattenedOptionalScopes } = processScopes(
-      requiredScopes,
-      optionalScopes,
-      { findNetworkClientIdByChainId },
+    assignAccountsToScopes(flattenedRequiredScopes, permittedAccounts);
+    assignAccountsToScopes(flattenedOptionalScopes, permittedAccounts);
+
+    const sessionScopes = mergeScopes(
+      flattenedRequiredScopes,
+      flattenedOptionalScopes,
     );
 
-    Object.keys(flattenedRequiredScopes).forEach((scope) => {
-      if (scope !== 'wallet') {
-        flattenedRequiredScopes[scope].accounts = permittedAccounts.map(
-          (account) => `${scope}:${account}`,
-        );
-      }
-    });
-    Object.keys(flattenedOptionalScopes).forEach((scope) => {
-      if (scope !== 'wallet') {
-        flattenedOptionalScopes[scope].accounts = permittedAccounts.map(
-          (account) => `${scope}:${account}`,
-        );
-      }
-    });
+    await Promise.all(
+      Object.keys(scopedProperties || {}).map(async (scopeString) => {
```

Comment:
so the source of truth for which networks to add will come from the returned confirmation. I don't think we even want to necessarily implement this as a stepping stone to naively take and add all scoped properties... But I suppose we can until we have a confirmation. At some point we may just want to create a little mock confirmation to have this logic more solid and testable